compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/FirDataFlowAnalyzer.kt

```diff
@@ -338,6 +339,16 @@ class FirDataFlowAnalyzer(private val components: FirAbstractBodyResolveTransfor
         graphBuilder.exitJump(jump).mergeIncomingFlow()
     }
 
+    // ----------------------------------- Check not null call -----------------------------------
+
+    fun exitCheckNotNullCall(checkNotNullCall: FirCheckNotNullCall) {
+        // Note: For DFA purposes, `arg!!` is equivalent to `(arg as TypeOfArg)`
+        val node = graphBuilder.exitCheckNotNullCall(checkNotNullCall).mergeIncomingFlow()
+        val type = checkNotNullCall.typeRef.coneTypeSafe<ConeKotlinType>()?.withNullability(ConeNullability.NOT_NULL) ?: return
```

Comment:
It's enough to add `Any` to set of possible types (`FirDataFlowInfo(setOf(session.builtinTypes.anyType.coneTypeUnsafe()), emptySet())`)

After smartcast there will be intersection type `T? & Any` that will be reduced to `T`

Reply:
Ah good to know! Done. Now I understand why that's done elsewhere (e.g., for `null` equality expressions).